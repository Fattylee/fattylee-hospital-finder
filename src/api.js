const createTodo = (todoPath = "/todos", todo) => {
  return new Promise((resolve, reject) => {
    if (todoPath.includes("todo")) {
      console.log("=============Real createTodo api=============");
      setTimeout(() => {
        console.log("========after 3s=========");
        return resolve(todo);
      });
    } else {
      reject("failed to include a todo in the route");
    }
  });
};

module.exports = { createTodo };
