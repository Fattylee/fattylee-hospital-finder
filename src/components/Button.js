import React, { useState } from "react";
import styled from "./Counter.module.css";

export const Button = ({ increment, onHandleClick }) => {
  const handleClick = () => {
    onHandleClick(increment);
  };

  return (
    <button className={styled.btn} onClick={handleClick}>
      {increment}+
    </button>
  );
};
