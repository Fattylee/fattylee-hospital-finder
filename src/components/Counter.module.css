.btn:hover {
  cursor: pointer;
}

.btn {
  margin: 10px;
}
