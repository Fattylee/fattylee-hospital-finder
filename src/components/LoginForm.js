import React, { useState } from "react";

export const LoginForm = (prop) => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");
  const [rememberme, setRememberme] = useState(true);

  const handleChange = (e) => {
    const { name, value, id, type, checked } = e.target;
    console.log(name, value, id, type, checked);

    if (name === "username") {
      setUsername(value);
      prop.onHandleUsername(value);
    } else if (name === "password") {
      setPassword(value);
      prop.onHandlePassword(value);
    } else if (name === "rememberme") {
      setRememberme(checked);
      prop.onHandleRememberme(checked);
    }
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    prop.onHandleSubmit(username, password, rememberme);
    const input = e.target.elements;
    // u can get form element by name or id value
    console.log(input["username"]);
    clearForm();
  };

  const clearForm = () => {
    setPassword("");
    setUsername("");
    setRememberme(false);
  };

  return (
    <div>
      <h2>{username}</h2>
      <form data-testid="login-form" onSubmit={handleSubmit}>
        <label htmlFor="username">Username</label>
        <input
          type="text"
          name="username"
          data-testid="username"
          id="username"
          value={username}
          onChange={handleChange}
        />
        <br />
        <label htmlFor="password">
          Password
          <input
            type="password"
            name="password"
            data-testid="password"
            id="password"
            value={password}
            onChange={handleChange}
          />
        </label>
        <br />
        <label htmlFor="rememberme">
          Remember me
          <input
            type="checkbox"
            name="rememberme"
            data-testid="rememberme"
            id="rememberme"
            checked={rememberme}
            onChange={handleChange}
          />
        </label>
        <button>Submit</button>
      </form>
    </div>
  );
};
