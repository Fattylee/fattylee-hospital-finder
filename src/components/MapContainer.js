import React, { Fragment } from "react";
import GoogleMapReact from "google-map-react";

const MapContainer = () => {
  const SG_COOR = { lat: "99", lng: "12345" };
  return (
    <Fragment>
      <h1>MapContainer Loaded</h1>
      <GoogleMapReact
        bootstrapURLKeys={{
          key: "AIzaSyAn6KqtG5IxmjoyGNY3Bh6KvLkXWZ5k45E",
          libraries: ["places", "directions"],
        }}
        defaultZoom={11}
        defaultCenter={{ lat: SG_COOR.lat, lng: SG_COOR.lng }}
        yesIWantToUseGoogleMapApiInternals={true}
        onGoogleApiLoaded={({ map, maps }) => this.apiHasLoaded(map, maps)} // "maps" is the mapApi. Bad naming but that's their library.
      >
        {/* Pin markers on the Map*/}
        {/*markers.map((marker, key) => {
        const { name, lat, lng } = marker;
        return <MapMarker key={key} name={name} lat={lat} lng={lng} />;
      })*/}
      </GoogleMapReact>
    </Fragment>
  );
};

export default MapContainer;
