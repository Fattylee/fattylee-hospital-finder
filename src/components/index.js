export { default as Products } from "./products/Products";
export { default as Landing } from "./Landling/Landing";
export { default as Header } from "./header/Header";
export { default as Login } from "./Login/Login";
export { default as Register } from "./Register/Register";
