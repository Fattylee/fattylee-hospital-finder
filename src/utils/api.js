import Axios from "axios";

export const setOrClearAuthToken = (token) => {
  if (token) {
    Axios.defaults.headers.common["Authorization"] = token;
  } else {
    delete Axios.defaults.headers.common["Authorization"];
  }
};
