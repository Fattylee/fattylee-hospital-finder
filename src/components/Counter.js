import React, { useState, useRef } from "react";
import styled from "./Counter.module.css";
import { Button } from "./Button";

export const Counter = ({ label = "nothing much" }) => {
  const [counter, setCounter] = useState(0);
  const handleIncrement = (amount) => setCounter(counter + amount);

  return (
    <div>
      <hr />

      <h2>{label}</h2>
      <h1>This is my counter App</h1>
      <h2>Count: {counter}</h2>
      <Button increment={1} onHandleClick={handleIncrement} />
      <Button increment={10} onHandleClick={handleIncrement} />
      <Button increment={100} onHandleClick={handleIncrement} />
      <Button increment={1000} onHandleClick={handleIncrement} />

      <div>
        <button className={styled.btn} onClick={() => setCounter(0)}>
          Clear counter
        </button>
      </div>
      <Todo todos={[{ id: Date.now(), todo: "Read the quran" }]} />
    </div>
  );
};

export function Todo({ todos: initialTodos = [{ id: 123, todo: "code!!" }] }) {
  const [todos, setTodos] = useState(initialTodos);
  const [todo, setTodo] = useState("");
  const inputRef = useRef(null);

  return (
    <div>
      <h2>Todo List:</h2>
      <ul>
        {todos.map((todo) => (
          <li key={todo.id}>{todo.todo}</li>
        ))}
      </ul>

      <form
        data-testid="form"
        onSubmit={(e) => {
          e.preventDefault();
          console.log("submitted");
          console.log(inputRef.current.value);
          const todo = inputRef?.current?.value.trim();

          if (todo) {
            setTodos((prevTodo) => [...prevTodo, { id: Date.now(), todo }]);

            // clear todo input : uncontrolled input
            inputRef.current.value = "";
            // clear todo input : controlled input
            setTodo("");
          }
        }}
      >
        <label htmlFor="message">Enter some message</label>
        <input
          type="text"
          id="message"
          data-testid="message"
          placeholder="Enter a todo"
          ref={inputRef}
          value={todo}
          onChange={(e) => setTodo(e.target.value)}
        />
      </form>
    </div>
  );
}
