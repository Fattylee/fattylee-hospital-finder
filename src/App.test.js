import React from "react";
import ReactDOM from "react-dom";
// import App from "./App";
import { Counter, Todo } from "./components/Counter";
import renderer from "react-test-renderer";
import { fireEvent, render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import api, { createTodo } from "./api";

describe("App", () => {
  test("should render App on the DOM", () => {
    const root = document.createElement("div");

    ReactDOM.render(<Counter />, root);

    expect(root.querySelector("div h1").textContent).toMatch(/counter app/i);
  });

  test("should render snapShot", () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render counter correctly using @testing-library/react", () => {
    const { getByText, getByLabelText } = render(<Counter />);
    const elem = getByText("This is my counter App");
    expect(elem).not.toBeNull();
    expect(elem.textContent).toMatch(/counter\s+app/i);
    expect(elem).toHaveTextContent(/counter\s*app/i);
    getByLabelText(/some message/);
    getByLabelText("Enter some message");
    getByText(/nothing Much/i);
    // expect(getByLabelText("Enter some message")).toHaveTextContent(
    //   "Enter some message"
    // );
  });

  test("should render counter again", () => {
    const { getByText } = render(<Counter label="this is serious" />);
    getByText("this is serious");
  });

  test("should click count button", () => {
    const { getByText } = render(<Counter />);
    getByText("Count: 0");
    const btnPlusOne = getByText("1+");
    fireEvent.click(btnPlusOne);
    fireEvent.click(btnPlusOne);
    fireEvent.click(getByText("10+"));
    getByText("Count: 12");
    fireEvent.click(getByText("100+"));
    getByText("Count: 112");
    // fireEvent.
  });
});

describe("Todo:", () => {
  test("should render Todo snapshot 1", () => {
    const { toJSON } = renderer.create(<Todo />);
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should render Todo snapshot 2", () => {
    const { toJSON } = renderer.create(
      <Todo todos={[{ id: 9876, todo: "fake and mock all" }]} />
    );
    const tree = toJSON();
    expect(tree).toMatchSnapshot();
  });

  test.skip("should add new todo correctly", async () => {
    // mock createTodo
    // const mockFunc = (api.createTodo = jest.fn());
    const mockFunc = jest.fn();

    const { getByText, getByTestId, getByLabelText, findByText } = render(
      <Todo todos={[{ id: Date.now(), todo: "write test" }]} />
    );

    getByText(/write tesT/i);
    const labelText = getByLabelText("Enter some message");
    fireEvent.change(labelText, { target: { value: "say goodbye" } });

    // waitFor(async () => {
    fireEvent.submit(getByTestId("form"));
    getByText("say goodbye");
    expect(mockFunc).toHaveBeenCalledTimes(30);
    expect(3).toBe(2);
    // })
    //   .then((res) => {
    //     console.log("==============res==========");
    //   })
    //   .catch((err) => {
    //     console.log("=================err=============");
    //   });

    // fireEvent.change(labelText, { target: { value: "recite ur quran" } });

    // expect(getByTestId("message")).toHaveAttribute(
    //   "placeholder",
    //   "Enter a todo"
    // );

    // expect(getByTestId("message")).toHaveAttribute("value", "recite ur quran");

    // fireEvent.submit(getByTestId("form"));
    // expect(getByTestId("message")).toHaveAttribute("value", "");
    // getByText("recite ur quran");
    // findByText("recite ur quran");
  });
});
