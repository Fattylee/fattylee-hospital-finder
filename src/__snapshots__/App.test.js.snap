// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render snapShot 1`] = `
<div>
  <hr />
  <h2>
    nothing much
  </h2>
  <h1>
    This is my counter App
  </h1>
  <h2>
    Count: 
    0
  </h2>
  <button
    className="btn"
    onClick={[Function]}
  >
    1
    +
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    10
    +
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    100
    +
  </button>
  <button
    className="btn"
    onClick={[Function]}
  >
    1000
    +
  </button>
  <div>
    <button
      className="btn"
      onClick={[Function]}
    >
      Clear counter
    </button>
  </div>
  <div>
    <h2>
      Todo List:
    </h2>
    <ul>
      <li>
        Read the quran
      </li>
    </ul>
    <form
      data-testid="form"
      onSubmit={[Function]}
    >
      <label
        htmlFor="message"
      >
        Enter some message
      </label>
      <input
        data-testid="message"
        id="message"
        onChange={[Function]}
        placeholder="Enter a todo"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;

exports[`Todo: should render Todo snapshot 1 1`] = `
<div>
  <h2>
    Todo List:
  </h2>
  <ul>
    <li>
      code!!
    </li>
  </ul>
  <form
    data-testid="form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="message"
    >
      Enter some message
    </label>
    <input
      data-testid="message"
      id="message"
      onChange={[Function]}
      placeholder="Enter a todo"
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`Todo: should render Todo snapshot 2 1`] = `
<div>
  <h2>
    Todo List:
  </h2>
  <ul>
    <li>
      fake and mock all
    </li>
  </ul>
  <form
    data-testid="form"
    onSubmit={[Function]}
  >
    <label
      htmlFor="message"
    >
      Enter some message
    </label>
    <input
      data-testid="message"
      id="message"
      onChange={[Function]}
      placeholder="Enter a todo"
      type="text"
      value=""
    />
  </form>
</div>
`;
